e2e/test_end_to_end.py

```diff
@@ -0,0 +1,2517 @@
+import os
+
+import intake
+import numpy as np
+import pytest
+
+from access_nri_intake.cli import build
+
+from .conftest import here
+
+"""
+args=Namespace(
+    config_yaml=[
+        '/scratch/tm70/ct1163/configs/cmip5.yaml', 
+        '/scratch/tm70/ct1163/configs/access-om2.yaml'],
+    build_base_path='/scratch/tm70/ct1163/test_cat/',
+    catalog_base_path='./',
+    catalog_file='metacatalog.csv',
+    version='v2024-11-18',
+    no_update=False
+    )
+"""
+
+
+def print_directory_tree(root, indent=""):
+    """
+    Pretty print a directory tree - code from chatgpt.
+    """
+    for item in os.listdir(root):
+        path = os.path.join(root, item)
+        if os.path.isdir(path):
+            print(f"{indent}├── {item}/")
+            print_directory_tree(path, indent + "│   ")
+        else:
+            print(f"{indent}├── {item}")
+
+
+@pytest.fixture(scope="session")
+def current_catalog():
+    """
+    Return the current catalog as an intake catalog.
+    """
+    metacat = intake.cat.access_nri
+    yield metacat
```

Comment:
What's the idea behind `yield`ing here instead of `return`ing? (Not an issue, just for my own edification.)